cognite/client/_api/transformations/schema.py

```diff
@@ -15,13 +15,21 @@ class TransformationSchemaAPI(APIClient):
     _RESOURCE_PATH = "/transformations/schema"
 
     def retrieve(
-        self, destination: TransformationDestination, conflict_mode: Optional[str] = None
+        self,
+        destination: TransformationDestination,
+        conflict_mode: Optional[str] = None,
+        with_instance_space: Optional[bool] = None,
+        is_connection_definition: Optional[bool] = None,
+        instance_type: Optional[str] = None,
```

Comment:
These parameters are only applicable for the `Instances` TransformationDestination? 

Then we should either add overloads to reflect that, or create a separate method.

Reply:
separate method is probably easier.

Reply:
Yes, applicable for `Instances `TransformationDestination. However, if I defined in separate method, then the request method will be different. 

Currently all the destination use the method; so that's the reason I used condition to check the 'Instances', 'nodes', and 'edges'

`TransformationSchemaAPI.retrieve(destination: cognite.client.data_classes.transformations.common.TransformationDestination, conflict_mode: Optional[str] = None) → cognite.client.data_classes.transformations.schema.TransformationSchemaColumnList`

Reply:
> if I defined in separate method, then the request method will be different.
Not following. Could you elaborate?

Let's just introduce a `retrieve_instances_schema` method

Reply:
All the destination types are use the `transformations.schema.retrieve` method; so defining separate method like for example  `transformations.schema.retrieve_instances` will not be a good opinion according to my understanding.  

I can try with overload option by providing multiple versions of the `retrieve `method;

```
def retrieve(self, destination: InstancesTransformationDestination) -> TransformationSchemaColumnList:
    # Implementation for Instances transformation destination
    pass

def retrieve(self, destination: TransformationDestination, conflict_mode: str) -> TransformationSchemaColumnList:
    # Implementation for other transformation destinations
    pass
```

Reply:
Unfortunately, In Python, method overloading is not supported in the same way as in statically typed languages.

I got below error in git workflow; however, the code is run local machine without any error. 

```
cognite/client/_api/transformations/schema.py:87: error: Name "retrieve" already defined on line 17  [no-redef]
        def retrieve(
        ^
```

If we defined new method to `retrieve` transformation schemas endpoint for Instance destination only (ex: `TransformationSchemaAPI.retrieve_instances)` then inconsistence break. Document https://cognite-sdk-python.readthedocs-hosted.com/en/latest/transformations.html#get-transformation-schema;

My take on this, the request `TransformationSchemaAPI.retrieve() `should have for all the schema endpoints. 

So, the only way is to define `retrieve `method as shown below;

```
def retrieve(
    self,
    destination: TransformationDestination,
    conflict_mode: Optional[str] = None,
    with_instance_space: Optional[bool] = None,
    is_connection_definition: Optional[bool] = None,
    instance_type: Optional[str] = None,
) -> TransformationSchemaColumnList:
```